### PR TITLE
Chapter 2 - RStudio GUI guidance

### PR DESCRIPTION
**Pull request by rmbielby**
Added in some text and images on the use of git within RStudio for those that prefer a GUI.

Primarily as a new section: 2.15 in the Local Project chapter. Went quite image heavy given it's orientated around using the GUI.

### CODE REVIEW

File: 02-localproject.Rmd [hunk 1]
```diff
@@ -343,21 +345,98 @@ As mentioned earlier, a .gitignore file was created with our R Project. You can
...
+
+![RStudio Clone Git Repository](www/images/chapter2_RStudio_Clone.PNG)
+
+Here you can enter your existing `Repository URL` and choose a root directory for where to keep it locally. The `Project directory name` option will be auto-filled once you provide a `Repository URL`. For the root directory where the repository directory will be cloned, we recommend not using your standard One Drive folder system as this can cause issues if both One Drive and git are trying to manage file changes. Instead it's best to navigate either to `C:\` or `C:\Users\<username>\` and create a suitable root folder (e.g. `my_repos`) to store your repositories in (and pinning this folder to Quick access in File Explorer can help with finding your repository directories and files quickly in Windows).
+
+Once you've entered your repository URL and chosen a directory in which to clone your repository, click `Create Project` and RStudio will do the rest.
+
+### Overview of the RStudio git GUIs
+
+There are three key areas within RStudio for accessing git functionality. The the Git Pane, the terminal and the RStudio menu bar.
```
Comment (tmorley1) [inline 02-localproject.Rmd:400]: Typo: the the

File: 02-localproject.Rmd [hunk 2]
```diff
@@ -343,21 +345,98 @@ As mentioned earlier, a .gitignore file was created with our R Project. You can
...
+
+#### Terminal ([Git BASH](#opening-git-bash))
+
+The terminal offers all the [Git BASH](#opening-git-bash) command line functionality already described. It can be accessed via the terminal pane usually found next to the R console pane.
+
+#### Git pane
+
+The Git pane offers an excellent quick look at the status of your repository, whilst offering quick links for viewing or committing changes, pulling and pushing, viewing the history, accessing the project settings, and creating and switching branches. Effectively all the common things you need to do with `Git`. The image below shows the git pane opened for a local clone of the repo for this guidance. It conssists of it's own menu bar with buttons for each of the options listed above, alongside a preview pane showing any files that have been modified, added or removed since your last commit.
```
Comment (tmorley1) [inline 02-localproject.Rmd:408]: Might be worth talking a bit about the "status" - what do the different letters mean?

File: 02-localproject.Rmd [hunk 3]
```diff
@@ -343,21 +345,98 @@ As mentioned earlier, a .gitignore file was created with our R Project. You can
...
+
+![RStudio Git Panel](www/images/chapter2_RStudio_gitpanel.PNG)
+
+The elements in the git pane menu bar offer the following options:
+
+* <img src="www/images/chapter2_RStudio_gitpaneDiff.PNG" alt="Git diff icon" height="16"/>  Open the [RStudio review changes window](#branches-committing-and-pushing);
```
Comment (tmorley1) [inline 02-localproject.Rmd:414]: Is it possible to make the images here a little bigger? Some are a bit difficult to see

File: 02-localproject.Rmd [hunk 4]
```diff
@@ -343,21 +345,98 @@ As mentioned earlier, a .gitignore file was created with our R Project. You can
...
 ```
 will no longer show the image!
 
 In non-R projects you can create your own .gitignore file in a text editor and save it in your project folder.
 
+## Working with git in RStudio
+
+### Overview
+
+Within RStudio, you can choose to manage your repository using the git bash terminal as in the previous sections, use the RStudio GUI elements or use a combination of both. Below we highlight the key things you need to know to get going with a git repository within R.
```
Comment (tmorley1) [inline 02-localproject.Rmd:370]: Make it clear what GUI stands for

File: 02-localproject.Rmd [hunk 5]
```diff
@@ -343,21 +345,98 @@ As mentioned earlier, a .gitignore file was created with our R Project. You can
...
+
+The Review Changes window consists of 3 panes, the top left shows a preview of all files that have been added, modified or deleted, the top right pane allows you to add a commit message and the last pane shows the git diff result for any file selected in the file preview pane.
+
+![RStudio Review changes window](www/images/chapter2_RStudio_ReviewChanges.PNG)
+
+To make a commit, review the changes in the diff pane by selecting each file in turn and click the Stage button in the Review changes menu bar (either with mulitple/all files selected at once or eac file selected indivudually). Then add your commit message and click the `Commit` button below the Commit message entry box.
+
+Finally, you can just click the `Push` button above the commit message entry box to sync your changes to the remote repository.
+
+You can switch branches by clicking on the current branch name in the Review changes menu bar, i.e. next to the History button.
```
Comment (tmorley1) [inline 02-localproject.Rmd:442]: Is it worth including info on how to revert to previous versions using RStudio GUI?

Comment (tmorley1) [inline 02-localproject.Rmd:442]: Maybe a final paragraph to sum up the advantages/disadvantages of using RStudio GUI. What can't be done via the GUI?

File: 02-localproject.Rmd (hunk 3 above)
Comment (rmbielby) [reply 02-localproject.Rmd:414]: Increased them all in size by 25%

File: 02-localproject.Rmd [hunk 6]
```diff
@@ -343,21 +345,98 @@ As mentioned earlier, a .gitignore file was created with our R Project. You can
...
+* <img src="www/images/chapter2_RStudio_gitpanePush.png" alt="Git push icon" height="16"/> Push the current commits;
+* <img src="www/images/chapter2_RStudio_gitpaneHistory.png" alt="Git history icon" height="16"/> Open the [RStudio review changes window](#branches-committing-and-pushing), defaulting to the History panel;
+* <img src="www/images/chapter2_RStudio_gitpaneProjectSettings.png" alt="Project seeings icon" height="16"/> Open the project settings window;
+* <img src="www/images/chapter2_RStudio_gitpaneNewBranch.png" alt="New branch icon" height="16"/> Create a new branch;
+* <img src="www/images/chapter2_RStudio_gitpaneSelectBranch.png" alt="Current branch" height="16"/> Shows the current branch and allows the user to switch branches via the drop down list;
+* <img src="www/images/chapter2_RStudio_gitpaneRefresh.png" alt="Refresh icon" height="16"/> Refresh the git pane preview.
+
+#### Menu bar access
+
+To access the git functionality via the menu bar, navigate to `Tools` > `Version control`. You should then see the options `Commit...`, `Pull Branches`, `Push Branch`, `Hsitory` and `Project Setup...`
```
[the remark this answers is not in the document]
Comment (rmbielby) [reply 02-localproject.Rmd:426]: Done

File: 02-localproject.Rmd (hunk 4 above)
Comment (rmbielby) [reply 02-localproject.Rmd:370]: done

File: 02-localproject.Rmd [hunk 7]
```diff
@@ -343,21 +345,98 @@ As mentioned earlier, a .gitignore file was created with our R Project. You can
...
+
+Within RStudio, you can choose to manage your repository using the git bash terminal as in the previous sections, use the RStudio GUI elements or use a combination of both. Below we highlight the key things you need to know to get going with a git repository within R.
+
+### Initial set-up
+
+RStudio needs a few small bits of set-up to be able to access git. The first step is to let it know where the git executable can be found.
+
+Using the menu bar in RStudio, go to `Tools` > `Global options...` You should see the window below pop up, where you can select `Git/SVN`.
+
+![RStudio Global options](www/images/chapter2_RStudio_GlobalOptions.PNG)
```
[the remark this answers is not in the document]
Comment (rmbielby) [reply 02-localproject.Rmd:378]: As far as I can tell, I have. I could expand the alt text to be more descriptive.

File: 02-localproject.Rmd (hunk 5 above)
Comment (rmbielby) [reply 02-localproject.Rmd:444]: Added a couple of references to this.

Comment (rmbielby) [reply 02-localproject.Rmd:444]: Added a short paragraph along these lines. Let me know if you think it needs anything extra.

**Review by tmorley1 — APPROVED** [review]

**Pull request merged**